components/bench/ycsb.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/magiconair/properties"
+	_ "github.com/pingcap/go-ycsb/db/tikv"
+	"github.com/pingcap/go-ycsb/pkg/client"
+	"github.com/pingcap/go-ycsb/pkg/measurement"
+	_ "github.com/pingcap/go-ycsb/pkg/workload"
+	"github.com/pingcap/go-ycsb/pkg/ycsb"
+	"github.com/spf13/cobra"
+	"runtime"
+	"strings"
+)
+
+type ycsbConfig struct {
+	PropertyFile   string
+	RecordCount    uint32
+	OperationCount uint32
+	ReadAllFields  bool
+
+	ReadProportion            float32
+	UpdateProportion          float32
+	ScanProportion            float32
+	InsertProportion          float32
+	ReadModifyWriteProportion float32
+
+	RequestDistribution string
+	Verbose             bool
+	Pd                  string
+	ConnCount           uint32
+	BatchSize           uint32
+}
+
+const commonWorkloadURL = "https://raw.githubusercontent.com/pingcap/go-ycsb/master/workloads/workload"
+
+func (config ycsbConfig) toProperties() *properties.Properties {
+	result := properties.NewProperties()
+	if config.PropertyFile != "" {
+		// "a" to "f" are some workloads that used a lot
+		// expand these shorthands to URL
+		if len(config.PropertyFile) == 1 {
+			for _, content := range []string{"a", "b", "c", "d", "e", "f"} {
+				if config.PropertyFile == content {
+					config.PropertyFile = commonWorkloadURL + content
+					break
+				}
+			}
+		}
+
+		if strings.HasPrefix(config.PropertyFile, "http") {
+			result = properties.MustLoadURL(config.PropertyFile)
```

Comment:
Does this mean if user passes a wrong url or file will panic?
prefer LoadURL and handle err